streaming/base/dataset.py

```diff
@@ -306,46 +303,14 @@ def __init__(self,
             stream.repeat = repeat
             stream.samples = pick
 
-        # Determine and distribute shuffle seed and shm prefix.
-        seed_rng = np.random.default_rng(shuffle_seed)
-        self.shuffle_seed = int(seed_rng.integers(1 << 60))
-        prefix_int = int(seed_rng.integers(1 << 24))
-        self._prefix = f'{prefix_int:06x}'
-
-        # Should be a unique shared directory per each StreamingDataset instantiation to avoid a
-        # conflict between a different StreamingDataset instance on a same machine.
-        start_time = time()
-        while True:
-            self._shared_dir = os.path.join(os.path.sep, 'tmp', 'streaming', self._prefix)
-            if os.path.exists(self._shared_dir):
-                prefix_int = int(seed_rng.integers(1 << 24))
-                self._prefix = f'{prefix_int:06x}'
-            else:
-                break
-            elapsed = time() - start_time
-            # Raise an exception if not finding a unique shared directory in 60 secs
-            if elapsed > 60:
-                raise RuntimeError(''.join([
-                    f'Could not find the unique shared directory, bailing out.',
-                    'Please provide a different `shuffle_seed` value.'
-                ]))
-
-            sleep(TICK)
-
-        # Initialize the distributed package and synchronize all the ranks
-        is_dist_pg_initialized = False
-        if self._rank_world.num_ranks > 1:
-            if dist.is_available() and not dist.is_initialized():
-                is_dist_pg_initialized = True
-                dist.init_process_group(backend='nccl' if torch.cuda.is_available() and
-                                        dist.is_nccl_available() else 'gloo',
-                                        rank=world.rank,
-                                        world_size=world.num_ranks)
-            dist.barrier()
+        # Register/lookup our shared memory prefix and filelock root directory.
+        my_locals = [os.path.abspath(stream.local) for stream in streams]
```

Comment:
what happens if `streams` is empty ?

Reply:
streaming dataset init requires there to be at least one stream (or remote/local which is converted into a stream)